Source/SwiftLintBuiltInRules/Rules/Metrics/NestingRule.swift

```diff
@@ -130,8 +134,12 @@ private enum SwiftStructureKind: Equatable {
     case expression(SwiftExpressionKind)
     case statement(StatementKind)
 
-    init?(_ structureKind: String) {
+    init?(_ structureKind: String, ignoreTypealiasAndAssociatedtype: Bool) {
         if let declarationKind = SwiftDeclarationKind(rawValue: structureKind) {
+            if ignoreTypealiasAndAssociatedtype,
```

Comment:
I don't think `SwiftStructureKind` should know about the option. You might want to check this outside of this initializer.

Reply:
I didn't think of it until you said it. Certainly, `SwiftStructureKind` should know about that.
And on second thought, I moved the checking outside and inside `.declaration` case before next `validate` methods.

Is the code as you expected...?